app/src/main/java/com/codingblocks/cbonlineapp/notifications/NotificationsActivity.kt

```diff
@@ -39,20 +41,22 @@ class NotificationsActivity : BaseCBActivity() {
                 url: String,
                 videoId: String
             ) {
-                notificationDao.updateseen(notificationID)
-                if (url.contains("courseRun", true) ||
-                    url.contains("classroom", true)
-                ) {
-                    Router.open("activity://courseRun/$url")
-                } else if (url.contains("player", true)) {
-                    val activityRoute =
-                        Router.getRoute("activity://courseRun/$url") as ActivityRoute
-                    activityRoute
-                        .withParams(VIDEO_ID, videoId)
-                        .open()
-                } else {
-                    this@NotificationsActivity.openChrome(url)
-                }
+                GlobalScope.async {
```

Comment:
why async ?

Reply:
To update record asynchronously so that main thread will not wait and it will execute URL Open functionality immedietly.

Reply:
I don't see any use of async here